pkg/scheduler/extender/extender.go

```diff
@@ -424,10 +424,9 @@ func (e *Extender) createReservation(ctx context.Context, namespace string, name
 
 	// fill in node requests
 	reservation.NodeRequests = &genV1.NodeRequests{}
-	if nodes, err := e.prepareListOfRequestedNodes(nodes); err == nil {
-		reservation.NodeRequests.Requested = nodes
-	} else {
-		return err
+	reservation.NodeRequests.Requested = e.prepareListOfRequestedNodes(nodes)
+	if len(reservation.NodeRequests.Requested) == 0 {
+		return fmt.Errorf("createReservation: list of requested nodes is empty, incoming nodes count: %d", len(nodes))
```

Comment:
Will this error be returned in `filter` function? Here we set error on FilterRequest response - https://github.com/dell/csi-baremetal/blob/81b899af27f794acbc9d0a6c9c787593a2fe74b9/pkg/scheduler/extender/extender.go#L125. After error kube-sceduler will not send requests anymore

Reply:
If there is no nodes how should filter behave ?

Reply:
In my opinion we should return just empty list, not error in main `FilterHandler`?

Reply:
@mishoyama any suggestion?

Reply:
No need to return error - we just don't have appropriate nodes for scheduling at the current moment.

Reply:
👍